vncviewer/MonitorIndicesParameter.cxx

```diff
@@ -0,0 +1,170 @@
+/* Copyright 2021 Hugo Lundin <huglu@cendio.se> for Cendio AB.
+ * 
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * 
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#include <algorithm>
+#include <vector>
+#include <set>
+
+#include <FL/Fl.H>
+#include <rfb/LogWriter.h>
+
+#include "MonitorIndicesParameter.h"
+
+using namespace rfb;
+static LogWriter vlog("MonitorIndicesParameter");
+
+MonitorIndicesParameter::MonitorIndicesParameter(const char* name_, const char* desc_, const char* v)
+: StringParameter(name_, desc_, v) {}
+
+std::set<int> MonitorIndicesParameter::get()
+{
+    int value = 0;
+    int count = 0;
+    std::set<int> indices;
+    std::set<int> config_indices;
+    std::vector<MonitorIndicesParameter::Monitor> monitors = this->monitors();
+
+    if (monitors.size() <= 0) {
+        vlog.error("No monitors found.");
+        return indices;
+    }
+
+    // sscanf will modify the parsed string. Therefore
+    // we make a copy of it before parsing.
+    const char* config = getValueStr();
+
+    while (*config) {
+        if (1 == sscanf(config, "%d%n", &value, &count)) {
```

Comment:
Syntax errors?

Reply:
The parsing is now more robust and more validation has been added.